arch/x86/core/ia32/excstub.S

```diff
@@ -106,18 +106,32 @@ SECTION_FUNC(TEXT, _exception_enter)
 	pushl	%ebx
 	pushl	%ebp
 
+	/*
+	 * Save  the current esp in eax to avoid ifdef the code
+	 * in case of gdbstub be enable
+	 */
+	leal (%esp), %eax
```

Comment:
can't we push the segment registers later and remove the need for this? you could move the block of code in `#ifdef CONFIG_GDBSTUB `to right after the `CONFIG_LAZY_FPU_SHARING` block on line 159

Reply:
If I not mistaken initially I had put after push `esp` to avoid it, you suggested move to where co-op where pushed. If I change it I will need to re-order the struct again. I'd like to keep it this way if you don't mind.

Reply:
no you don't put this after you push esp. pushing esp places the esf as the argument to the C code that gets called.

>  If I change it I will need to re-order the struct again. I'd like to keep it this way if you don't mind.

I don't think we should intentionally introduce inefficiencies in our exception handling code so I'd rather this just got moved. This doesn't seem like a complicated change to me?

Reply:
I was referring the `esp` before interruption occurred, pushed before `CONFIG_LAZY_FPU_SHARING`, it is a parameter in `nanoEsf`. I don't mind change it, that was basically my initial approach, to be precise it was before `CONFIG_LAZY_FPU_SHARING` but it does not seem to change anything. I was just getting confuse with it. I can avoid that instruction adding more `ifdef` but it would make it more complicated to read.